src/page/post_view/video_loader.rs

```diff
@@ -103,7 +110,56 @@ pub fn VideoView(idx: usize, muted: RwSignal<bool>) -> impl IntoView {
         Some(())
     });
 
+    #[cfg(feature = "hydrate")]
+    {
+        let (watched_percentage, set_watched_percentage) = create_signal(0_u8);
+        let (watched_count, set_watched_count) = create_signal(0_u8);
+
+        let _ = use_event_listener(container_ref, ev::timeupdate, move |event| {
+            let target = event.target().unwrap();
+            let video = target.unchecked_into::<web_sys::HtmlVideoElement>();
+            let duration = video.duration();
+            let current_time = video.current_time();
+
+            set_watched_percentage.update(|watched_percentage| {
+                *watched_percentage = (100.0 * (current_time / duration)) as u8;
+            });
+
+            if current_time == 0.0 {
+                set_watched_count.update(|count| *count += 1);
+            }
+        });
+
+        let post = move || video_queue.get_untracked().get(idx).cloned();
```

Comment:
use `Signal::derive(move || video_queue.with(|q| q.get(idx).cloned()))`
you're cloning the whole list

Reply:
How is this cloning the whole list? Cloned will close the particular post, right?

Reply:
video_queue.get_untracked clones the underlying object
use with or with_untracked to access by reference